release-tools/stochss_ami_manager.py

```diff
@@ -0,0 +1,619 @@
+                                            stderr=self.stderr,
+                                            shell=True)
+            self.process.communicate()
+            if self.verbose: print 'End of cmd $', self.cmd
+
+        thread = threading.Thread(target=target)
+        thread.start()
+
+        if timeout is not None:
+            thread.join(timeout)
+            if thread.is_alive():
+                if silent is False:
+                    print 'Terminating process due to timeout...'
+                self.process.terminate()
+                thread.join()
+                if silent is False:
+                    print 'Process return code =', self.process.returncode
+        else:
+            thread.join()
+            if self.process.returncode != 0:
+                raise ShellCommandException("return code = {0}".format(self.process.returncode))
+
+
+
+class AmiManager:
+    INVALID_INSTANCE_TYPES = ['t1.micro', 'm1.small', 'm1.medium', 'm3.medium']
+    NUM_TRIALS = 5
+    NUM_SSH_TRIALS = 10
+    DEFAULT_AMI_FILENAME = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'ami.json')
+
+    def __init__(self, options):
+        self.uuid = uuid.uuid4()
+        self.base_ami_id = options['base_ami_id']
+        self.aws_region = options['aws_region']
+
+        if options['instance_type'] in AmiManager.INVALID_INSTANCE_TYPES:
+            print 'Instance type "{0}" is not sufficient to create StochSS AMI! Exiting.'.format(
+                options['instance_type'])
+            sys.exit(1)
+
+        self.instance_type = options['instance_type']
+
+        self.dependencies = options['dependencies']
+        self.python_packages = options['python_packages']
+        self.git_repo = options['git_repo']
+
+        print 'GIT REPO: {0}'.format(self.git_repo['url'])
+        print 'BRANCH: {0}'.format(self.git_repo['branch'])
+
+        self.ec2_connection = self.create_ec2_connection(aws_region=self.aws_region)
+
+        self.key_name = "test_stochss_kp_{0}".format(self.uuid)
+        self.key_file = None
+        self.security_groups = None
+        self.instance_ip = None
+        self.instance_id = None
+        self.instance_user = options['instance_user']
+
+        if "log" in options.keys():
+            self.log_type = options["log"]["type"]
+        else:
+            self.log_type = "screen"
+
+        if self.log_type == "file":
+            stdout_log_filename = os.path.join(os.path.dirname(os.path.abspath(__file__)), options["log"]["stdout"])
+            print 'stdout_log_filename: {0}'.format(stdout_log_filename)
+            self.stdout_log = open(stdout_log_filename, 'w')
+
+            stderr_log_filename = os.path.join(os.path.dirname(os.path.abspath(__file__)), options["log"]["stderr"])
+            print 'stderr_log_filename: {0}'.format(stderr_log_filename)
+            self.stderr_log = open(stderr_log_filename, 'w')
+
+        self.is_old_ami = False
```

Comment:
This code and the function below should probably just get removed since we are not using it anymore.  This will help maintenance in the future, and we have the git history if it is needed.